misk/src/main/kotlin/misk/web/jetty/WsConnections.kt

```diff
@@ -0,0 +1,36 @@
+package misk.web.jetty
+
+import com.squareup.moshi.Moshi
+import java.util.concurrent.ConcurrentHashMap
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class WsConnections {
+    @Inject lateinit var moshi: Moshi
+    private val mapping = ConcurrentHashMap<String, MutableSet<Socket>>()
+
+    fun subscribe(
+            socket: Socket,
+            topic: String
+    ) {
+        if (mapping[topic] == null) {
```

Comment:
This pattern is not thread safe. Use ``mapping.computeIfAbsent()`` to ensure that two requests arriving at the same time don't overwrite the topic's socket set on initial creation.